### ISSUE TITLE
Mypy reports error when using NoReturn

### ISSUE THREAD

**Issue by JakobDev** [python/mypy#12550]
**Bug Report**

According to the [Python documentation](https://docs.python.org/3/library/typing.html#typing.NoReturn), NoReturn indicating that a function never returns, but Mypy shows a error when a function with NoReturn has no return statement.

**To Reproduce**

Use Mypy on this code:

```python
from typing import NoReturn


def test() -> NoReturn:
    print("Hello World!")
```

**Expected Behavior**

Mypy should show no error. This function is marked which NoReturn, what means is has no return statement. This function actually has no return statement. So everything is correct here.

**Actual Behavior**

Mypy reports:
```
test.py:4: error: Implicit return in function which does not return
```

It complains, that a function which should have no return statement has no return statement.

**Your Environment**

Mypy version: 0.942

**Comment by TeamSpen210** [comment]
That’s not what NoReturn means. It indicates that the function will never successfully return to the caller, instead entering an infinite loop, always raising an exception, stopping the process etc.

**Comment by AlexWaygood** [comment]
@TeamSpen210 is correct.
